File: src/main/java/com/back/together02be/users/repository/UsersRepository.java
package com.back.together02be.users.repository;

import org.springframework.data.jpa.repository.JpaRepository;

import com.back.together02be.users.entity.Users;

import java.util.Optional;

public interface UsersRepository extends JpaRepository<Users, Long> {
    Optional<Users> findByUsername(String username);
    Optional<Users> findByRefreshToken(String refreshToken);
    boolean existsByUsername(String username);
}

File: src/main/java/com/back/together02be/users/repository/UsersRepository.kt
package com.back.together02be.users.repository

import com.back.together02be.users.entity.Users
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UsersRepository : JpaRepository<Users, Long> {
    fun findByUsername(username: String): Optional<Users>
    fun findByRefreshToken(refreshToken: String): Optional<Users>
    fun existsByUsername(username: String?): Boolean
}